modules/product/src/ProductTestTrait.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\commerce_product;
+
+use Drupal\commerce_product\Entity\ProductAttribute;
+use Drupal\commerce_product\Entity\ProductAttributeValue;
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Provides methods for working with products, variations, and attributes.
+ *
+ * This trait is meant to be used only by test classes.
+ *
+ * @todo Move to \Drupal\Tests\commerce_product post-SimpleTest.
+ */
+trait ProductTestTrait {
+
+  /**
+   * Creates a set of product cariations.
+   *
+   * The $values_matrix parameter contains the price and attributes information.
+   *
+   * Example:
+   * [
+   *   [
+   *      'price' => 9.99
+   *      'attribute_color' => 2,
+   *      'attribute_size' => 10,
+   *   ],
+   *   [
+   *      'price' => 9.99
+   *      'attribute_color' => 2,
+   *      'attribute_size' => 6,
+   *   ],
+   * ]
+   *
+   * @param string $type
+   *   The variation type.
+   * @param array $values_matrix
+   *   Array of values.
+   * @param string $currency_code
+   *   The currency code.
+   *
+   * @return \Drupal\commerce_product\Entity\ProductVariationInterface[]
+   *   Array of created variations
+   */
+  protected function createProductVariations($type = 'default', $values_matrix = [], $currency_code = 'USD') {
+    $variations = [];
+    foreach ($values_matrix as $item) {
+
+      $variation_values = [
+        'type' => $type,
+        'sku' => strtolower($this->randomMachineName()),
+        'price' => [
+          'currency_code' => $currency_code,
+        ],
+      ];
+
+      foreach ($item as $key => $value) {
+        if ($key == 'price') {
+          $variation_values['price']['amount'] = $value;
+        }
+        else {
+          $variation_values[$key] = $value;
+        }
+      }
+
+      $variation = ProductVariation::create($variation_values);
+      $variation->save();
+      $variations[] = ProductVariation::load($variation->id());
+    }
+
+    return $variations;
+  }
+
+  /**
+   * Creates an attribute field and set of attribute values.
+   *
+   * @param string $variation_type
+   *   The variation type.
+   * @param string $name
+   *   The attribute field name.
+   * @param array $options
+   *   Associative array of key name values. [red => Red].
+   * @param bool $test_field
+   *   Flag to create a test field on the attribute.
+   *
+   * @return \Drupal\commerce_product\Entity\ProductAttributeValueInterface[]
+   *   Array of attribute entities.
+   */
+  protected function createAttributeSet($variation_type, $name, array $options, $test_field = FALSE) {
```

Comment:
This should just be called createAttributeSet(), since it only creates a single attribute, not several.